test/e2e/local-cluster/create_cluster.go

```diff
@@ -0,0 +1,229 @@
+	cmd = exec.Command("kind", args...)
+	_, err = cmd.Output()
+	if err != nil {
+		l.Error(err, "kubeconfig export")
+		os.Exit(1)
+	}
+
+	l.Info("kubeconfig exported successfully")
+
+	// Build image
+	image := fmt.Sprintf("%s:%s", dockerImageRepository, dockerImageTag)
+	if !dockerImageBuildSkip {
+		err = dockerImageBuild("./../../../", image)
+		if err != nil {
+			l.Error(err, "image build")
+			os.Exit(1)
+		}
+		l.Info("Docker image built successfully")
+	}
+
+	// Load image
+	err = cluster.LoadImage(ctx, image)
+	if err != nil {
+		l.Error(err, "image load")
+		os.Exit(1)
+	}
+	l.Info("Docker image loaded to the cluster successfully")
+
+	// Deploy Greenhouse manager
+	err = installChart(ctx, "./../../../charts/manager", greenhouseControllerManagerRelease, kubeconfig, greenhouseControllerManagerNamespace)
+	if err != nil {
+		l.Error(err, "deploy greenhouse")
+		os.Exit(1)
+	}
+	l.Info("Greenhouse manager is deployed successfully")
+
+}
+
+func installChart(ctx context.Context, dir, release, kubeconfig string, namespace string) error {
+
+	l := log.FromContext(ctx)
+
+	chart, err := loader.Load(dir)
+	if err != nil {
+		return err
+	}
+
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(
+		&genericclioptions.ConfigFlags{
+			KubeConfig: &kubeconfig,
+			Namespace:  &namespace,
+		},
+		namespace,
+		"secret",
+		l.V(10).Info,
+	); err != nil {
+		return err
+	}
+
+	// These values are the minimal set of required values to deploy manager helm chart
+	globalDNS := map[string]interface{}{"dnsDomain": "greenhouse.cloudoperators"}
+	chart.Values["global"] = globalDNS
+
+	controllerManagerImage := map[string]interface{}{"repository": dockerImageRepository, "tag": dockerImageTag}
+	controllerManager, ok := chart.Values["controllerManager"].(map[string]interface{})
+	if !ok {
+		l.Error(err, "value merge")
+		os.Exit(1)
+	}
+	controllerManager["image"] = controllerManagerImage
+	controllerManager["replicas"] = "1"
+	chart.Values["controllerManager"] = controllerManager
+	// These values are the minimal set of required values to deploy manager helm chart
+
+	get := action.NewGet(actionConfig)
+	_, err = get.Run(release)
+	if err != nil {
+		if err.Error() == "release: not found" {
+			client := action.NewInstall(actionConfig)
+			client.ReleaseName = release
+			client.Namespace = namespace
+			client.CreateNamespace = true
+			client.Wait = true
+			client.Timeout = TEST_TIMEOUT
+
+			if _, err := client.Run(chart, chart.Values); err != nil {
+				l.Error(err, "chart install")
+				return err
+			}
+		}
+	} else {
+		client := action.NewUpgrade(actionConfig)
+		client.Namespace = namespace
+		client.Wait = true
+		client.Timeout = TEST_TIMEOUT
+		if _, err := client.Run(release, chart, chart.Values); err != nil {
+			l.Error(err, "chart upgrade")
+			return err
+		}
+	}
+
+	return nil
+}
+
+func dockerImageBuild(path string, repoAndtag string) error {
```

Comment:
The image building does not work for me... 
If I build `greenhouse:e2e-latest` manually and run
```
$ go run . --dockerImageBuildSkip true
I0703 10:39:04.543662   71429 create_cluster.go:63] "configuration loaded" kindClusterName="greenhouse-e2e" dockerImageRepository="greenhouse" dockerImageTag="e2e-latest" dockerImageBuildSkip=true kubeconfigName="e2e.kubeconfig" kubeconfigInternalName="e2e.internal.kubeconfig"
I0703 10:40:34.759070   71429 create_cluster.go:80] "cluster created successfully"
I0703 10:40:35.001250   71429 create_cluster.go:108] "kubeconfig exported successfully"
I0703 10:40:39.955923   71429 create_cluster.go:127] "Docker image loaded to the cluster successfully"
I0703 10:41:20.487547   71429 create_cluster.go:135] "Greenhouse manager is deployed successfully"
```
I get a running controller (see remark about missing Dex CRDs).


But without a manually built image the make target fails.

1. Logging sais image was build successfully
2. Logging errors on loading the image to kind cluster
3. I cannot find the image on my machine (grepping for `e2e` in `docker images` is empty)

```
$ docker rmi greenhouse:e2e-latest
Untagged: greenhouse:e2e-latest
Deleted: sha256:a311d6cfdfd36e3e57c70884d2ab98d31c4fd6c9b1aec8c1fe2f984d6958da87

$ make e2e-local-cluster-create
cd test/e2e/local-cluster && go run .
I0703 11:21:21.821216   86111 create_cluster.go:63] "configuration loaded" kindClusterName="greenhouse-e2e" dockerImageRepository="greenhouse" dockerImageTag="e2e-latest" dockerImageBuildSkip=false kubeconfigName="e2e.kubeconfig" kubeconfigInternalName="e2e.internal.kubeconfig"
I0703 11:21:33.310028   86111 create_cluster.go:80] "cluster created successfully"
I0703 11:21:33.545322   86111 create_cluster.go:108] "kubeconfig exported successfully"
I0703 11:21:48.355849   86111 create_cluster.go:118] "Docker image built successfully"
E0703 11:21:48.377069   86111 create_cluster.go:124] "image load" err="kind: load docker-image greenhouse:e2e-latest failed: exit status 1: exit status 1"
exit status 1
make: *** [Makefile:180: e2e-local-cluster-create] Error 1
$ docker images | grep e2e
```

Reply:
I have added another function to extract error messages from Docker build response by 1ad6ac7b8a592a1d454abc3befc271718272c7c8

Tested with the same approach:

```
> docker rmi greenhouse:e2e-latest
Untagged: greenhouse:e2e-latest
Deleted: sha256:25c652679b8980fa778c238f60ca76cd17bed3f000fe0056bc37227a4cf94b52

> make e2e-local-cluster-create
cd test/e2e/local-cluster && go run .
I0709 18:12:49.448102 1291840 create_cluster.go:69] "configuration loaded" kindClusterName="greenhouse-e2e" dockerImageRepository="greenhouse" dockerImageTag="e2e-latest" dockerImageBuildSkip=false kubeconfigName="e2e.kubeconfig" kubeconfigInternalName="e2e.internal.kubeconfig"
I0709 18:12:49.637200 1291840 create_cluster.go:86] "cluster created successfully"
I0709 18:12:49.883694 1291840 create_cluster.go:114] "kubeconfig exported successfully"
I0709 18:12:50.911422 1291840 create_cluster.go:124] "Docker image built successfully"
I0709 18:12:51.042014 1291840 create_cluster.go:133] "Docker image loaded to the cluster successfully"

 > docker image ls | grep e2e-latest
greenhouse                                                         e2e-latest                                 25c652679b89   About a minute ago   253MB

Reply:
This was resolved by calling the **docker** cli directly